### PR TITLE
Convert TrashbinActivity to Kotlin

### PR DESCRIPTION
**Pull request by alperozturk96**
- [x] Tests written, or not not needed

### CODE REVIEW

**Review by tobiasKaminsky — APPROVED** [review]

File: app/src/main/java/com/owncloud/android/ui/trashbin/TrashbinActivity.kt [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+        }
+
+        binding = TrashbinActivityBinding.inflate(layoutInflater)
+
+        setContentView(binding.root)
+        setupToolbar()
+
+        findViewById<View>(R.id.sort_list_button_group).visibility = View.VISIBLE
+        findViewById<View>(R.id.switch_grid_view_button).visibility =
+            View.GONE
```
Comment (ZetaTom) [inline app/src/main/java/com/owncloud/android/ui/trashbin/TrashbinActivity.kt:118]:
```suggestion
        findViewById<View>(R.id.switch_grid_view_button).visibility = View.GONE
```

File: app/src/main/java/com/owncloud/android/ui/trashbin/RemoteTrashbinRepository.kt [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        @Deprecated("Deprecated in Java")
+        override fun doInBackground(vararg voids: Void?): Boolean {
+            return try {
+                val client = clientFactory.create(user)
+                val result = RemoveTrashbinFileRemoteOperation(file!!.fullRemotePath)
+                    .execute(client)
+                result.isSuccess
+            } catch (e: CreationException) {
+                Log_OC.e(this, "Cannot create client", e)
+                java.lang.Boolean.FALSE
```
Comment (ZetaTom) [inline app/src/main/java/com/owncloud/android/ui/trashbin/RemoteTrashbinRepository.kt:67]:
```suggestion
                false
```

All occurrences of `java.lang.Boolean.*` could be replaced with `true` and `false` respectively.

File: app/src/main/java/com/owncloud/android/ui/trashbin/RemoteTrashbinRepository.kt [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        @Deprecated("Deprecated in Java")
+        override fun doInBackground(vararg voids: Void?): Boolean {
+            return try {
+                val client = clientFactory.createNextcloudClient(user)
+                val emptyTrashbinFileOperation = EmptyTrashbinRemoteOperation()
+                val result = emptyTrashbinFileOperation.execute(client)
+                result.isSuccess
+            } catch (e: CreationException) {
+                Log_OC.e(this, "Cannot create client", e)
+                java.lang.Boolean.FALSE
```
Comment (ZetaTom) [inline app/src/main/java/com/owncloud/android/ui/trashbin/RemoteTrashbinRepository.kt:97]:
```suggestion
                false
```

File: app/src/main/java/com/owncloud/android/ui/trashbin/RemoteTrashbinRepository.kt [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                val client = clientFactory.create(user)
+                val result = ReadTrashbinFolderRemoteOperation(remotePath).execute(client)
+                if (result.isSuccess) {
+                    trashbinFiles = result.resultData
+                    true
+                } else {
+                    false
+                }
+            } catch (e: CreationException) {
+                java.lang.Boolean.FALSE
```
Comment (ZetaTom) [inline app/src/main/java/com/owncloud/android/ui/trashbin/RemoteTrashbinRepository.kt:167]:
```suggestion
                false
```

**Review by ZetaTom — COMMENTED** [review]
In my brief tests, this change worked well. The requested changes are minor.

File: app/src/main/java/com/owncloud/android/ui/trashbin/RemoteTrashbinRepository.kt (hunk 2 above)
Comment (alperozturk96) [reply app/src/main/java/com/owncloud/android/ui/trashbin/RemoteTrashbinRepository.kt:67]: IDE automatically added those, I missed it, will be fixed in next commit thanks

**Pull request merged**